Providers/Profile.swift

```diff
@@ -499,7 +499,18 @@ open class BrowserProfile: Profile {
         let commands = items.map { item in
             SyncCommand.displayURIFromShareItem(item, asClient: id)
         }
-        return self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
+        return self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> {
+            let syncStatus = self.syncManager.syncClients()
+            syncStatus >>> {
+                let deviceIDs = clients.filter({ $0.fxaDeviceId != nil }).map({ $0.fxaDeviceId! })
+                guard let account = self.getAccount() else {
+                    return
+                }
+                
+                account.notify(deviceIDs: deviceIDs, collectionsChanged: ["clients"])
+            }
+            return syncStatus
+        }
```

Comment:
This block needs tidying. 

Do we need to `notify` explicitly after `syncClients()` has returned? If so, comment.

Also, there's a `Deferred` inside a `Deferred`, is there anyway to tidy this?